File: apps/api-core/internal/dto/audit.go
package dto

import (
	"encoding/json"
	"time"
)

type AuditLogResponse struct {
	ID            uint            `json:"id"`
	EntityType    string          `json:"entityType"`
	EntityID      uint            `json:"entityId"`
	Action        string          `json:"action"`
	OldValues     json.RawMessage `json:"oldValues,omitempty"`
	NewValues     json.RawMessage `json:"newValues,omitempty"`
	ChangedFields json.RawMessage `json:"changedFields,omitempty"`
	UserID        *uint           `json:"userId,omitempty"`
	Username      string          `json:"username,omitempty"`
	CreatedAt     time.Time       `json:"createdAt"`
}

type AuditLogQuery struct {
	EntityType string `form:"entityType"`
	StartDate  string `form:"startDate"`
	EndDate    string `form:"endDate"`
	Action     string `form:"action"`
	UserID     *uint  `form:"userId"`
	EntityID   *uint  `form:"entityId"`
}

type AuditLogListResponse struct {
	ID            uint      `json:"id"`
	EntityType    string    `json:"entityType"`
	EntityID      uint      `json:"entityId"`
	Action        string    `json:"action"`
	ChangedFields []string  `json:"changedFields"`
	UserID        *uint     `json:"userId,omitempty"`
	Username      string    `json:"username,omitempty"`
	CreatedAt     time.Time `json:"createdAt"`
}

File: apps/frontend-web/src/api/v1/audit.ts
import { api } from "boot/axios";
import { ListResponse, PageRequestParams, SortRequestParams } from "src/schema/response";
import { HistoryType } from "src/schema/revision";

export interface AuditLog {
  id: number;
  entityType: string;
  entityId: number;
  action: HistoryType;
  changedFields: string[];
  userId: number;
  username: string;
  createdAt: string;
}

export interface AuditLogDetail extends AuditLog {
  oldValues: Record<string, unknown> | null;
  newValues: Record<string, unknown> | null;
}

export type FetchAuditLogsRequestParams = Partial<
  {
    entityType: string;
    startDate: string;
    endDate: string;
    action: HistoryType;
    userId: number;
    entityId: number;
  } & PageRequestParams &
    SortRequestParams
>;

export async function fetchAuditLogs(params: FetchAuditLogsRequestParams) {
  const result = await api.get<ListResponse<AuditLog>>("/api/v1/admin/audit-logs", {
    params,
  });
  return result.data;
}

export async function fetchAuditLogDetail(id: number) {
  const result = await api.get<{ value: AuditLogDetail }>(`/api/v1/admin/audit-logs/${id}`);
  return result.data.value;
}
